### PR TITLE
Remove Connection.getVersion

### PR DESCRIPTION
**Pull request by thecoop**
Remove the `Connection.getVersion` property. For the moment, redirect instances to `getNode().getVersion()`, to be tidied up in subsequent steps

### CODE REVIEW

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-1 elasticsearch-ci/part-3

**Comment by thecoop** [comment]
@elasticsearchmachine rerun elasticsearch-ci/part-3

File: server/src/main/java/org/elasticsearch/transport/TcpTransport.java [hunk 1]
```diff
@@ -263,13 +262,6 @@ public final class NodeChannels extends CloseableConnection {
             compressionScheme = connectionProfile.getCompressionScheme();
         }
 
-        @Override
-        public Version getVersion() {
-            // TODO: this should be the below, but in some cases the node version does not match the passed-in version.
-            // return node.getVersion();
-            return Version.fromId(version.id());
```
[the remark this answers is not in the document]
Comment (thecoop) [reply server/src/main/java/org/elasticsearch/transport/TcpTransport.java:270]: This implementation is currently incorrect - this should have been converted from TransportVersion a while ago, but was left out. This value isn't persisted somewhere, so we should be able to just remove it.

**Review by ldematte — APPROVED** [review]
LGTM

**Review by DaveCTurner — APPROVED** [review]
LGTM (assuming CI is happy anyway, I've not studied the test changes in enormous detail)

**Pull request merged**